third_party/ObjexxFCL/src/ObjexxFCL/vectorize.hh

```diff
@@ -23,16 +23,33 @@
 #define ASSUME(b)
 #define ASSUME_ALIGNED(p,b) p=(decltype(p))__builtin_assume_aligned(p,b)
 #define RESTRICT __restrict__
-#elif defined(_WIN32) && ( defined(_MSC_VER) || defined(__INTEL_COMPILER) )
+
+#elif defined(_WIN64) && ( defined(_MSC_VER) || defined(__INTEL_COMPILER) )
 #define ASSUME(b) __assume(b)
 #define ASSUME_ALIGNED(p,b) __assume_aligned(p,b)
 #define RESTRICT __restrict
+
+// Disable alignment hints for 32-bit windows builds
+#elif defined(_WIN32) && ( defined(_MSC_VER) || defined(__INTEL_COMPILER) )
+#define ASSUME(b) __assume(b)
+#define ASSUME_ALIGNED(p,b) 
+#define RESTRICT __restrict
+
 #else
 #define ASSUME(b)
 #define ASSUME_ALIGNED(p,b)
 #define RESTRICT
 #endif
 
+// Alignment for Compiler Options
+#if defined (__AVX2__)
```

Comment:
@mbadams5 is right that AVX2 doesn't provide the 512 bit types: I didn't mean to imply that. I've got mixed information on whether 64 byte alignment is helpful with AVX2 but 32 byte alignment is probably sufficient so we can change that top category from `__AVX2__` to `__MIC__` for now: MIC is for Intel Phi h/w but we'll want 64 byte alignment for AVX-512 (aka AVX3) but there is no predefined macro for __AVX512__ or __AVX3__ yet. We need to do our own experiments: it may be that 64 byte is best anyway since it lines up with the cache line size. A little over-alignment is pretty harmless -- a few wasted bytes per array.

What I'm doing for the next ObjexxFCL release is setting the default alignment but allowing a larger alignment override so that it can be tuned for each application.